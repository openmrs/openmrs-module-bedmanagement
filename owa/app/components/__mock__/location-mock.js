Object.defineProperties(window.location, {
    host : {
        writable: true,
        value: '192.168.33.10'
    },
    hostname : {
        writable: true,
        value: '192.168.33.10'
    },
    href :{
        writable: true,
        value: 'https://192.168.33.10/openmrs/owa/bedmanagement/admissionLocations.html'
    },
    origin :{
        writable: true,
        value: 'https://192.168.33.10'
    },
    pathname :{
        writable: true,
        value: '/openmrs/owa/bedmanagement/admissionLocations.html'
    },
    port :{
        writable: true,
        value: ''
    },
    protocol :{
        writable: true,
        value: 'https:'
    }
});