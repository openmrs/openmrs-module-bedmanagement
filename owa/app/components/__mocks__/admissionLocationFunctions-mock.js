import pickBy from 'lodash.pickby';
import {admissionLocations, visitLocations, bedTypes} from 'utilities/__tests__/testData.json';

const admissionLocationFunctionsMock = {
    setActiveLocationUuid: (admissionLocationUuid) => jest.fn(),
    getActiveLocationUuid: () => {
        return 'baf7bd38-d225-11e4-9c67-080027b662ec';
    },
    setState: (data) => jest.fn(),
    getState: () => jest.fn(),
    getAdmissionLocations: () => {
        return admissionLocations;
    },
    getVisitLocations: () => {
        return visitLocations;
    },
    getBedTypes: () => {
        return bedTypes;
    },
    getAdmissionLocationByUuid: (admissionLocationUuid) => {
        return typeof admissionLocations[admissionLocationUuid] !== 'undefined' ?
            admissionLocations[admissionLocationUuid] : null;
    },
    getParentAdmissionLocation: (admissionLocationUuid) => {
        const parentAdmissionLocationUuid = admissionLocations[admissionLocationUuid].parentAdmissionLocationUuid;
        return parentAdmissionLocationUuid != null ? admissionLocations[parentAdmissionLocationUuid] : null;
    },
    getChildAdmissionLocations : (admissionLocationUuid) => {
        return  pickBy(admissionLocations, (admissionLocation) => {
            return admissionLocation.parentAdmissionLocationUuid == admissionLocationUuid;
        });
    },
    reFetchAllAdmissionLocations : () => jest.fn(),
    notify : (notifyType, message) => jest.fn()
};

export default admissionLocationFunctionsMock;
