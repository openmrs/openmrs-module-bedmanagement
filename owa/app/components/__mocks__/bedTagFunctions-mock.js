import _ from 'lodash';

import {bedTags} from 'utilities/__tests__/testData.json';

const bedTagFunctionsMock = {
    setState: (stateData) =>  jest.fn(),
    getBedTags: () => {
        return bedTags;
    },
    getBedTagByUuid: (bedTagUuid) => {
        return _.find(bedTags, function (bedTag) {
            return bedTag.uuid == bedTagUuid;
        });
    },
    fetchBedTags: () => jest.fn(),
    notify : (notifyType, message) => jest.fn()
};

export default bedTagFunctionsMock;
