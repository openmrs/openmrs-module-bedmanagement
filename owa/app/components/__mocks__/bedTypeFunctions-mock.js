import _ from 'lodash';

import {bedTypes} from 'utilities/__tests__/testData.json';

const bedTypeFunctionsMock = {
    setState: (stateData) =>  jest.fn(),
    getBedTypes: () => {
        return bedTypes;
    },
    getBedTypeById: (bedTypeId) => {
        return _.find(bedTypes, function (bedType) {
            return bedType.id == bedTypeId;
        });
    },
    getBedTypeName: (bedTypeName) => {
        return _.find(bedTypes, function (bedType) {
            return bedType.name == bedTypeName;
        });
    },
    fetchBedTypes: () => {
        return bedTypes;
    },
    notify : (notifyType, message) => jest.fn()
};

export default bedTypeFunctionsMock;
