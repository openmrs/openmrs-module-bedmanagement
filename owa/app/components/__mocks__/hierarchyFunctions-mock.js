import {admissionLocations} from 'utilities/__tests__/testData.json';

const hierarchyFunctionMock = {
    getActiveUuid: () => {
        return 'baf7bd38-d225-11e4-9c67-080027b662ec';
    },
    getAdmissionLocations: () => {
        return admissionLocations;
    },
    setActiveUuid: (locationUuid) => jest.fn(),
    setState: (stateData) => jest.fn(),
    toggleIsOpen: (locationUuid, isOpen) => jest.fn(),
    setAdmissionLocationIsOpen: (locationUuid, isOpen) => jest.fn()
};

export default hierarchyFunctionMock;