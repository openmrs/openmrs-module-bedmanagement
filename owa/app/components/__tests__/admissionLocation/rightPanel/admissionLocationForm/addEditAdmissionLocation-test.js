import React from 'react';
import {shallow, mount} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

import AddEditAdmissionLocation from 'components/admissionLocation/rightPanel/admissionLocationForm/addEditAdmissionLocation';
import admissionLocationFunctionsMock from 'components/__mocks__/admissionLocationFunctions-mock';

require('components/__mocks__/location-mock');
const testProps = {
    admissionLocationFunctions: admissionLocationFunctionsMock,
    admissionLocations: admissionLocationFunctionsMock.getAdmissionLocations()
};

describe('AddEditAdmissionLocation', () => {

    beforeAll(() => {
        var mock = new MockAdapter(axios);
        const data = {
            'ward': {
                'uuid': 'baf7bd38-d225-11e4-9c67-080027b662ec',
                'display': 'General Ward',
                'name': 'General Ward',
                'description': 'Hospital General Ward Section',
            }
        };

        mock.onPost('https://192.168.33.10/openmrs/ws/rest/v1/admissionLocation/baf7bd38-d225-11e4-9c67-080027b662ec')
            .reply(200, data);
    });

    it('Should display admission location form properly', () => {
        const addAdmissionLocationForm = shallow(<AddEditAdmissionLocation activeUuid='baf7bd38-d225-11e4-9c67-080027b662ec'
            admissionLocationFunctions={testProps.admissionLocationFunctions} operation='add' />);
        expect(addAdmissionLocationForm.find('span').text().trim()).toBe('General Ward');
        expect(addAdmissionLocationForm.find('input[type=\'text\']').props().value).toBe('');
        expect(addAdmissionLocationForm.find('textarea').props().value).toBe('');
        expect(shallowToJson(addAdmissionLocationForm)).toMatchSnapshot();

        const editAdmissionLocationForm = shallow(<AddEditAdmissionLocation activeUuid='e48fb2b3-d490-11e5-b193-0800270d80ce'
            admissionLocationFunctions={testProps.admissionLocationFunctions} operation='edit' />);
        expect(editAdmissionLocationForm.find('span').text().trim()).toBe('General Ward');
        expect(editAdmissionLocationForm.find('input[type=\'text\']').props().value).toBe('General Ward - Room 2');
        expect(editAdmissionLocationForm.find('textarea').props().value).toBe('General Ward room number 2');
        expect(shallowToJson(editAdmissionLocationForm)).toMatchSnapshot();

        const editTopLevelAdmissionLocationForm = shallow(<AddEditAdmissionLocation activeUuid='baf7bd38-d225-11e4-9c67-080027b662ec'
            admissionLocationFunctions={testProps.admissionLocationFunctions} operation='edit' />);
        expect(editTopLevelAdmissionLocationForm.find('select').props().value).toBe('c1e42932-3f10-11e4-adec-0800271c1b75');
        expect(editTopLevelAdmissionLocationForm.find('input[type=\'text\']').props().value).toBe('General Ward');
        expect(editTopLevelAdmissionLocationForm.find('textarea').props().value).toBe('Hospital General Ward Section');
        expect(shallowToJson(editTopLevelAdmissionLocationForm)).toMatchSnapshot();
    });

    it('Should trigger event handler', () => {
        const spyOnCancelEventHandler = jest.spyOn(AddEditAdmissionLocation.prototype, 'cancelEventHandler');
        const spyOnSubmitHandler = jest.spyOn(AddEditAdmissionLocation.prototype, 'onSubmitHandler');
        const spyOnSelectParentLocation = jest.spyOn(AddEditAdmissionLocation.prototype, 'onSelectParentLocation');
        const spyOnChangeDescriptionField = jest.spyOn(AddEditAdmissionLocation.prototype, 'onChangeDescriptionField');
        const spyOnChangeNameField = jest.spyOn(AddEditAdmissionLocation.prototype, 'onChangeNameField');
        const spySetState = jest.spyOn(testProps.admissionLocationFunctions, 'setState');
        const editTopLevelAdmissionLocationForm = mount(<AddEditAdmissionLocation activeUuid='baf7bd38-d225-11e4-9c67-080027b662ec'
            admissionLocationFunctions={testProps.admissionLocationFunctions} operation='edit' />);

        editTopLevelAdmissionLocationForm.find('input[type=\'text\']').simulate('change');
        expect(spyOnChangeNameField).toHaveBeenCalled();

        editTopLevelAdmissionLocationForm.find('textarea').simulate('change');
        expect(spyOnChangeDescriptionField).toHaveBeenCalled();

        editTopLevelAdmissionLocationForm.find('select').simulate('change');
        expect(spyOnSelectParentLocation).toHaveBeenCalled();

        editTopLevelAdmissionLocationForm.find('.btn-danger').simulate('click');
        expect(spyOnCancelEventHandler).toHaveBeenCalled();
        expect(spySetState).toHaveBeenCalledWith({
            activePage: 'listing',
            pageData: {},
            activeUuid: null
        });

        editTopLevelAdmissionLocationForm.find('input[type=\'submit\']').simulate('submit');
        expect(spyOnSubmitHandler).toHaveBeenCalled();
    });

});