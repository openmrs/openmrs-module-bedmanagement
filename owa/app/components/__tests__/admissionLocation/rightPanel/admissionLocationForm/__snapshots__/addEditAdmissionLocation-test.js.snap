// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddEditAdmissionLocation Should display admission location form properly 1`] = `
<div
  className="main-container"
>
  <div
    className="admission-location-form"
  >
    <div
      className="block-title"
    >
      Add
       Ward
      <a
        className="back-link"
        href="javascript:void(0);"
        onClick={[Function]}
        title="back"
      >
        &lt;&lt; Back
      </a>
    </div>
    <div
      className="block-content"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-block"
        >
          <label
            className="form-title"
          >
            Parent Location
          </label>
          <span>
            General Ward
          </span>
        </div>
        <div
          className="form-block"
        >
          <label
            className="form-title"
          >
            Name
          </label>
          <input
            onChange={[Function]}
            required={true}
            type="text"
            value=""
          />
        </div>
        <div
          className="form-block"
        >
          <label
            className="form-title"
          >
            Description
          </label>
          <textarea
            cols="50"
            name="description"
            onChange={[Function]}
            rows="4"
            value=""
          />
        </div>
        <div
          className="form-block"
        >
          <input
            className="btn btn-primary float-left margin-right"
            disabled={false}
            name="submit"
            type="submit"
            value="Save"
          />
          <input
            className="btn btn-danger float-left"
            name="submit"
            onClick={[Function]}
            type="button"
            value="Cancel"
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`AddEditAdmissionLocation Should display admission location form properly 2`] = `
<div
  className="main-container"
>
  <div
    className="admission-location-form"
  >
    <div
      className="block-title"
    >
      Edit
       Ward
      <a
        className="back-link"
        href="javascript:void(0);"
        onClick={[Function]}
        title="back"
      >
        &lt;&lt; Back
      </a>
    </div>
    <div
      className="block-content"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-block"
        >
          <label
            className="form-title"
          >
            Parent Location
          </label>
          <span>
            General Ward
          </span>
        </div>
        <div
          className="form-block"
        >
          <label
            className="form-title"
          >
            Name
          </label>
          <input
            onChange={[Function]}
            required={true}
            type="text"
            value="General Ward - Room 2"
          />
        </div>
        <div
          className="form-block"
        >
          <label
            className="form-title"
          >
            Description
          </label>
          <textarea
            cols="50"
            name="description"
            onChange={[Function]}
            rows="4"
            value="General Ward room number 2"
          />
        </div>
        <div
          className="form-block"
        >
          <input
            className="btn btn-primary float-left margin-right"
            disabled={false}
            name="submit"
            type="submit"
            value="Save"
          />
          <input
            className="btn btn-danger float-left"
            name="submit"
            onClick={[Function]}
            type="button"
            value="Cancel"
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`AddEditAdmissionLocation Should display admission location form properly 3`] = `
<div
  className="main-container"
>
  <div
    className="admission-location-form"
  >
    <div
      className="block-title"
    >
      Edit
       Ward
      <a
        className="back-link"
        href="javascript:void(0);"
        onClick={[Function]}
        title="back"
      >
        &lt;&lt; Back
      </a>
    </div>
    <div
      className="block-content"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-block"
        >
          <label
            className="form-title"
          >
            Parent Location
          </label>
          <select
            name="parent-location"
            onChange={[Function]}
            value="c1e42932-3f10-11e4-adec-0800271c1b75"
          >
            <option
              value=""
            >
              None
            </option>
            <option
              key="c1e42932-3f10-11e4-adec-0800271c1b75"
              value="c1e42932-3f10-11e4-adec-0800271c1b75"
            >
              Ganiyari
            </option>
            <option
              key="c1e4950f-3f10-11e4-adec-0800271c1b75"
              value="c1e4950f-3f10-11e4-adec-0800271c1b75"
            >
              Subcenter 2 (SEM)
            </option>
            <option
              key="c1f25be5-3f10-11e4-adec-0800271c1b75"
              value="c1f25be5-3f10-11e4-adec-0800271c1b75"
            >
              Subcenter 1 (BAM)
            </option>
          </select>
        </div>
        <div
          className="form-block"
        >
          <label
            className="form-title"
          >
            Name
          </label>
          <input
            onChange={[Function]}
            required={true}
            type="text"
            value="General Ward"
          />
        </div>
        <div
          className="form-block"
        >
          <label
            className="form-title"
          >
            Description
          </label>
          <textarea
            cols="50"
            name="description"
            onChange={[Function]}
            rows="4"
            value="Hospital General Ward Section"
          />
        </div>
        <div
          className="form-block"
        >
          <input
            className="btn btn-primary float-left margin-right"
            disabled={false}
            name="submit"
            type="submit"
            value="Save"
          />
          <input
            className="btn btn-danger float-left"
            name="submit"
            onClick={[Function]}
            type="button"
            value="Cancel"
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;
